t/api/02-iso.t

```diff
@@ -1297,4 +1301,73 @@ subtest '_SKIP_CHAINED_DEPS prevents scheduling parent tests' => sub {
     $schema->txn_rollback;
 };
 
+subtest 'schedule tests correctly when changing TEST to job template name' => sub {
+    $schema->txn_begin;
+    add_opensuse_test('parent', JOB_TEMPLATE_NAME => 'parent_variant1');
+    add_opensuse_test(
+        'child',
+        START_AFTER_TEST  => 'parent',
+        JOB_TEMPLATE_NAME => 'child_variant1'
+    );
+    add_opensuse_test(
+        'child',
+        NOT_ADD_TESTSUITE => 1,
+        START_AFTER_TEST  => 'parent',
+        JOB_TEMPLATE_NAME => 'child_variant2'
+    );
+
+    my $schedule_param = {
+        ISO       => $iso,
+        DISTRI    => 'opensuse',
+        VERSION   => '13.1',
+        FLAVOR    => 'DVD',
+        ARCH      => 'i586',
+        BUILD     => '0091',
+        _GROUP_ID => '1002'
+    };
+    my $res = schedule_iso($schedule_param);
```

Comment:
I now just created https://github.com/os-autoinst/openQA/pull/2562 for this so we do not need to duplicate all the same default settings everywhere. If you agree we can merge my PR and then please use instead:

```perl
my $res = schedule_iso(%iso, _GROUP_ID => '1002');
```

of course you can still define `$schedule_param` as a copy to reuse it later or you write e.g.:

```perl
schedule_iso(%iso, TEST => 'child');
```

and such

Reply:
After your PR has been merged, I will change this code.

Reply:
merged